src/Application/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -11,24 +11,32 @@
 
 declare(strict_types=1);
 
-namespace Sylius\PriceHistoryPlugin\Application\Provider;
+namespace Sylius\PriceHistoryPlugin\Application\Calculator;
 
 use Sylius\Component\Core\Exception\MissingChannelConfigurationException;
-use Sylius\Component\Core\Model\ChannelInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
 use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Webmozart\Assert\Assert;
 
-final class ProductVariantPriceProvider implements ProductVariantPriceProviderInterface
+final class ProductVariantPriceCalculator implements ProductVariantPriceCalculatorInterface
```

Comment:
I wonder if it wouldn't be better to decorate the `ProductVariantPriceCalculator` from Sylius with adding this new method, or change the name of this class to sth more focused on the lowest price, eg. `ProductVariantLowestPriceCalculator`

Reply:
I don't see any point in decorating it, since all we do is add a new method.
👍 for changing the name though.